internal/objects/reconciler.go

```diff
@@ -451,14 +473,17 @@ func (r *Reconciler) syncPropagated(inst, srcInst *unstructured.Unstructured) (s
 		!reflect.DeepEqual(canonical(inst), canonical(srcInst)) ||
 		inst.GetLabels()[api.LabelInheritedFrom] != srcInst.GetNamespace() {
 		metadata.SetLabel(inst, api.LabelInheritedFrom, srcInst.GetNamespace())
+
+		log.Info("src object doesn't exist creating", "namespace", ns.Name(), "namespaceParent", ns.Parent(), "inst", inst.GetName())
 		return actionWrite, srcInst
 	}
 
+	log.Info("src object exists on api server but not in forest", "namespace", ns.Name(), "namespaceParent", ns.Parent(), "inst", inst.GetName())
 	// The object already exists and doesn't need to be updated. This will typically happen when HNC
 	// is restarted - all the propagated objects already exist on the apiserver. Record that it exists
 	// for our statistics.
+	r.syncSource(log, inst)
```

Comment:
Makes sure the forest is up to date with everything that's been reconciled, especially important on restarts when the forest is new